esrally/driver/runner.py

```diff
@@ -1569,6 +1573,180 @@ def __repr__(self, *args, **kwargs):
+
+        It expects a parameter dict with the following mandatory keys:
+
+        * ``transform-id``: the transform id to start, the transform must have been created upfront.
+
+        The following keys are optional:
+        * ``force``: forcefully stop a transform, default false
+        * ``wait-for-checkpoint``: whether to wait until all data has been processed till the next checkpoint, default true
+        * ``wait-for-completion``: whether to block until the transform has stopped, default true
+        * ``transform-timeout``: overall runtime timeout of the transform in seconds, default 1800 (1h)
+
+        Returned meta data
+        `
+        The following meta data are always returned:
+
+        * ``search-time-ms``: The amount of time spent searching, in milliseconds.
+        * ``processing-time-ms``: The amount of time spent processing results, in milliseconds.
+        * ``index_time-ms``: The amount of time spent indexing, in milliseconds.
+        * ``weight``: operation-agnostic representation of the processed documents (used internally by Rally for throughput calculation).
+        * ``unit``: The unit in which to interpret, always "docs".
+        * ``ops``: A dictionary of counters, see below.
+
+        ``ops`` contains the following meta data:
+
+        * ``pages-processed``: The number of search or bulk index operations processed.
+        * ``documents-processed``: The number of documents that have been processed from the source index of the transform.
+        * ``documents-indexed``: The number of documents that have been indexed into the destination index for the transform.
+        * ``index-total``: The number of index operations on the dest index for the transform.
+        * ``index-failures``: The number of indexing failures.
+        * ``search-total``: The number of search operations on the source index for the transform.
+        * ``search-failures``: The number of search failures.
+        * ``processing-total``: The number of processing operations.
+
+        """
+        import time
+
+        transform_id = mandatory(params, "transform-id", self)
+        force = params.get("force", False)
+        timeout = params.get("timeout")
+        wait_for_completion = params.get("wait-for-completion", True)
+        wait_for_checkpoint = params.get("wait-for-checkpoint", True)
+        transform_timeout = params.get("transform-timeout", 60.0 * 60.0)
+
+        if not self._start_time:
+            self._start_time = time.time()
+            await es.transform.stop_transform(transform_id=transform_id,
+                                              force=force,
+                                              timeout=timeout,
+                                              wait_for_completion=False,
+                                              wait_for_checkpoint=wait_for_checkpoint)
+
+        stats_response = await es.transform.get_transform_stats(transform_id=transform_id)
```

Comment:
Does it make sense to add  await asyncio.sleep(wait_period)? with wait_period parameter and some default, otherwise we keep calling get_transform_stats constantly, which could affect the performance? Kind of like you had a `poll_interval` in your previous iteration.

Reply:
I was expecting rally is doing some rate limiting. Are you sure, rally isn't?

Reply:
:+1: I reintroduced `poll_interval`.